### PR TITLE
Add packaging to cluster stats response

### PR DESCRIPTION
**Pull request by jasontedor**
This commit adds a packaging_types field to the cluster stats response that outlines the build flavors and types present in a cluster.

Relates #39378

### CODE REVIEW

File: rest-api-spec/src/main/resources/rest-api-spec/test/cluster.stats/10_basic.yml [hunk 1]
```diff
@@ -71,3 +71,15 @@
...
+"get cluster stats returns packaging types":
+
+  - skip:
+      version: " - 7.99.99"
+      reason:  "packaging types are added for v8.0.0"
+
+  - do:
+      cluster.stats: {}
+
+  - is_true: nodes.discovery_types
```
Comment (jakelandis) [inline rest-api-spec/src/main/resources/rest-api-spec/test/cluster.stats/10_basic.yml]: should this be `packaging_types` ?

Comment (jasontedor) [inline rest-api-spec/src/main/resources/rest-api-spec/test/cluster.stats/10_basic.yml]:
```suggestion
  - is_true: nodes.packaging_types
```

Comment (jasontedor) [reply rest-api-spec/src/main/resources/rest-api-spec/test/cluster.stats/10_basic.yml]: Good catch. I pushed.

**Review by jakelandis — APPROVED** [review]
LGTM ... and `var` !!

**Comment by jasontedor** [comment]
You best believe it, and be ready!

**Pull request merged**